CONTRIBUTION.md

```diff
@@ -0,0 +1,44 @@
+# Code style and naming conventions
+
+We use `flake8` as a linter-check with `pep8-naming` conventions. 
+So, we use `variable_name`, `ClassName`, and `method_name`. For global variables, we use `UPPER_CASE_NAMES`.
+Sometimes, we the convention can be broken, but the rule should be written in the end to allow vscode to ignore it.
+
+We use type hints and `typing` library for hints like `List[int]`.
+
+We inherit `collections.abc` classes if we need to construct a collection with an extended behavior.
+
+We are using automatic testing with `pytest` please consult [Scripts/tests/README.md]. We use both unit and regression testing, however the coverage of code with unit-tests remains relatevly low.
+
+# Repository rules
+
+1. Please create an issue, and then pull-request, not vice versa.
+2. Please keep main branch in your fork updated so that it can be rebased (we prefere rebasing more than merging).
+3. Once you address an issue, please refere it in commit message by using phrases like 'Partially fixes #000'.
```

Comment:
"refere" -> "refer to"

"in commit" -> "in the commit"

"by using phrases like" -> "by a phrase such as"